### ISSUE TITLE
 [Feat]: Add the ability to modify properties/schema of existing Notion databases

### ISSUE THREAD

**Issue by Spiral-Memory** [RocketChat/Apps.Notion#61]
### What do we need?

Currently, there is no option to add or change properties of a Notion database after its creation. This option is only available during the database setup process. To modify the schema later, users must manually make changes within the Notion page itself, which can be inconvenient, especially when considering the benefits of integrations. The Notion API does offer functionality for property modification, which we can use to implement this feature.

### Acceptance Criteria

- [ ] Create a modal for selecting and adding/changing the database schema.
- [ ] Integrate this feature using the Notion API.

**Comment by Spiral-Memory** [comment]
@Nabhag8848 , I believe this feature would give users more control over their Notion databases within the Rc app. Please let me know if you also agree that this functionality would be valuable? If so, I'd be happy to begin working on its implementation.

**Comment by Nabhag8848** [comment]
@Spiral-Memory There is much more priority feature we needed earlier, Updating the Record which is created in DB through Create Page or Record Modal. 

lets say org is keeping track of project they are working on - in Notion Database and there is extension of project deadline, or it needs to assign to different person, xyz, etc. we can't do it currently which is needed and happens often. 

Would you like to work on this right now or you want to stick to ViewErrors which you were working on ? , i can open a issue and provide all the details in issue on what we want? It will take lot of time to merge (like complete march) cause its huge ?

- This Feature is Okay, but not so useful or priority, as this is not what user would use often, if you would think deeply, most of them won't prefer to use this even.

**Comment by Nabhag8848** [comment]
@Spiral-Memory Also this limitation will make this feature more unusable.
<img width="665" alt="This" src="https://github.com/RocketChat/Apps.Notion/assets/65061890/bd62228b-c9b3-4b02-a7b0-d2e7628fc23a">

**Comment by Spiral-Memory** [comment]
> @Spiral-Memory There is much more priority feature we needed earlier, Updating the Record which is created in DB through Create Page or Record Modal.
> 
> lets say org is keeping track of project they are working on - in Notion Database and there is extension of project deadline, or it needs to assign to different person, xyz, etc. we can't do it currently which is needed and happens often.
> 
> Would you like to work on this right now or you want to stick to ViewErrors which you were working on ? , i can open a issue and provide all the details in issue on what we want? It will take lot of time to merge (like complete march) cause its huge ?
> 
> * This Feature is Okay, but not so useful or priority, as this is not what user would use and need often, if you would think deeply, most of them won't prefer to use this even.

Sure, @Nabhag8848 , I understand. You can open the issue you are talking about along with ViewErrors. In parallel, I can try working on it. No problem if it takes time; if I can fix it, I will be happy to do so.

Yes, the feature I mentioned is not very useful. Maybe we can work on it later for the sake of making the Notion app more complete thats it.

**Comment by Spiral-Memory** [comment]
> @Spiral-Memory Also this limitation will make this feature [ modify Schema of DB ] more unusable. <img alt="This" width="665" src="https://private-user-images.githubusercontent.com/65061890/309533407-bd62228b-c9b3-4b02-a7b0-d2e7628fc23a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0NTExODEsIm5iZiI6MTcwOTQ1MDg4MSwicGF0aCI6Ii82NTA2MTg5MC8zMDk1MzM0MDctYmQ2MjIyOGItYzliMy00YjAyLWE3YjAtZDJlNzYyOGZjMjNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAzVDA3MjgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjZThiMDYwMzgyOGE2ODllMjYzMjc5ZjBjYmJhYzc0YTk5MGZjMDBlMmY2MjkwOWNiNzk4ZWNjNWYzNTc1ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fLuCfysubxsyjh3smDTHQWzihqhoxKb_km9P3-WYIQU">

Yep Actually, I tried doing this as well. I felt there are too many complexity to achieve it, have to take care of many cases, also the API had many limitation. Let's keep this issue open, after the priority features have been implemented, we may look into this at later stages

**Comment by Nabhag8848** [comment]
> > @Spiral-Memory Also this limitation will make this feature [ modify Schema of DB ] more unusable. <img alt="This" width="665" src="https://private-user-images.githubusercontent.com/65061890/309533407-bd62228b-c9b3-4b02-a7b0-d2e7628fc23a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk0NTExODEsIm5iZiI6MTcwOTQ1MDg4MSwicGF0aCI6Ii82NTA2MTg5MC8zMDk1MzM0MDctYmQ2MjIyOGItYzliMy00YjAyLWE3YjAtZDJlNzYyOGZjMjNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAzVDA3MjgwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjZThiMDYwMzgyOGE2ODllMjYzMjc5ZjBjYmJhYzc0YTk5MGZjMDBlMmY2MjkwOWNiNzk4ZWNjNWYzNTc1ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fLuCfysubxsyjh3smDTHQWzihqhoxKb_km9P3-WYIQU">
> 
> Yep Actually, I tried doing this as well. I felt there are too many complexity to achieve it, have to take care of many cases, also the API had many limitation. Let's keep this issue open, after the priority features have been implemented, we may look into this at later stages

- Will open issue, later today. until then watch this clickup app which does similar thing, the way it has been implemented i feel its not the best way, one of main reason when we need to update two or more props we need to have lot of clicks and Everytime need to select action and update it.  But What do you think about this ? 

https://youtu.be/8L_man2A1-M?feature=shared&t=76

- we were planning to implement just have a button on the message that has been send on creation - `update`. 

Open the Modal with All the input Elements with initial value, similar modal we already have in create record feature right ? and update those fields which needed to be in one go in input elements the one which are not updated in the input fields doesn't matter cause we can make the request of changed value and default which wasn't changed and update the fields. 
this allows to make multiple fields change in one go. 

What do you think ?

**Comment by Spiral-Memory** [comment]
Yes, @Nabhag8848 , you are correct. There can be cases when we want to update multiple things in one go. This ClickUp implementation will require updating things individually, which is not the best way. Yes, I have seen the create record implementation, and we can utilize a similar approach to implement this as well. My only concern, on which I am not sure, is whether the API we will be using supports receiving multiple data at once. I will go through that. The rest, I do agree with you.
